### ISSUE TITLE
Padding and truncation

### ISSUE THREAD

**Issue by vinbl** [helicalAI/helical#218]
Hello, I noticed that the tokenizers have `tokenizer.padding_side` set to `'left'` while the `tokenizer.truncation_side` is set to `'right'`. Is this the intended behavior? What is the reason? Thanks!

**Comment by mattwoodx** [comment]
Hi @vinbl,
I assume you are referring to the Helix-mRNA and Mamba2-mRNA models, since the other issues you opened are around these. Yes this is expected behaviour. Since both Helix-mRNA and Mamba2-mRNA are at their core SSM's, we use the default padding strategy for these models, which is left padding. The attention mask ensures the signal from these tokens is suppressed. Truncation and padding are mutually exclusive on a single sequence, so when a sequence goes beyond the maximum length specified, in this case 12288, the remaining nucleotides are removed and no padding takes place.

Here is a link to [Mamba2](https://huggingface.co/docs/transformers/en/model_doc/mamba2) on Hugging Face which speaks about the padding side.

**Comment by vinbl** [comment]
You also mention the mask - the tokenizer has two options: `return_special_tokens_mask` and `return_attention_mask`. Is only the former used, or should I generate and pass both when calling the model? Thank you!

**Comment by mattwoodx** [comment]
You only need `return_attention_mask`. For special tokens you just need to insert the letter 'E' within your sequence to denote codon separation. The tokenizer will handle the rest.
